v1.0.0

al1img opened the pull request.

al1img committed:
[um] Remove build dependencies as they are in meta-aos now

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[domd] Mount /var rw partition in initramfs

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[domd] Mount machine-id in initramfs

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[domd] Add origin xt components

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[um] Change renesasota revision to support new UM sources

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[domd] Remove libseccomp recipe and add meta-security layer

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[domd] Update crun and runc to latest revisions

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[nvme-cli] Fix nvme-cli RO rootfs issues

nvme-cli uses postinst script to generate IDs etc. RO rootfs doesn't support
this script. Move creating nvme IDs to nvme systemd service.

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[dom0] Move customizing of domd.service to domd-override.conf

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[dom0] Start domd after xenconsoled

On reboot, domd can't stop successfully because xenconsoled is already stopped.
Add xenconsoled dep to domd.service in order to stop domd after xenconsoled.

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[iam] Add create parameter to setup disk script

Setup script API was changed in meta-aos 6.0.1. Update input parameters
accordingly.

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[yaml] Add prefix AOS_ to all Aos related variables

To do not mixup Aos vars with XT vars, add AOS_ prefix.

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[yaml] Divide AOS_BOARD_MODEL to AOS_BOARD_MODEL and AOS_BOARD_VERSION

Currently, we use board model in the following format: "model;version".
We use different format of board model and board version in different places:
For example, for component ID we use: "model-version-component". The origin
combined AOS_BOARD_MODEL is divided into two part to make possibility to use
different format.

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[um] Add board model and board version to components ID

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[aos-update] Remove board model from bundle metadata

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[dom0] Remove shutting down all domains from guestreboot.sh

Now domains are stopped properly. Remove WA from guestreboot.sh.

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[README.md] Update with related to release 1.0.0 changes

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[TODO] Remove done items

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
Bump v1.0.0

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
Merge remote-tracking branch 'xen-troops/master' v0.8.5

Squashed commit of the following:

commit e8252a58e36d19618737ffd43f6466b3446cc140
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri Sep 23 23:57:46 2022 +0300

    README.md: update readme for 0.8.5 release

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 476bed95f8856f810763e0d377f646d5b6809f7c
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri Sep 23 23:56:06 2022 +0300

    README.md: re-arrange order of sections

    Move Networking SR-IOV section close to NVME SR-IOV section.

    Remove trailing whitespaces.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit cc2229eef486101cd9a9eab7b4ca4e564c0e1d75
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri Sep 23 23:52:14 2022 +0300

    xen: bump branch to spider-0.8.5

    We have added some changes to Xen since version 0.8.2, bump up branch
    to get those new changes into the build.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 0a0ced301712d5c5e95b4a91132a8d97fe9ff02b
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri Sep 23 02:02:26 2022 +0300

    domd, domu: bump linux kernel to v5.10.41/rcar-5.1.7.rc3-xt

    As Renesas released a new kernel version, update our kernel from
    v5.10.41/rcar-5.1.7.rc2-xt to v5.10.41/rcar-5.1.7.rc3-xt

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit bdd29d2e26260de0608a24d97ce981a78257b8a5
Author: Dmytro Semenets <dmytro_semenets@epam.com>
Date:   Mon Sep 19 20:35:18 2022 +0300

    README: Describe how to use SR-IOV for Intel PCIe NIC

    Describe how to configure SR-IOV feature on Ethernet controller.
    Ethernet Controller 10G X550T used as example device.

    Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
    Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit a1eb581652316fb87560d437167ec8deda0fd3ab
Author: Dmytro Semenets <dmytro_semenets@epam.com>
Date:   Mon Sep 19 18:16:09 2022 +0300

    domd: xen.dts: add one more IP-MMU context for PCIe0

    Add context for BFD 2:10:0 and onward to facilitate SR-IOV virtual
    functions of Intel 10G X505T ethernet controller.

    Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
    Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 2caa77ec5909c5e4f1d22689a856558dd3bdf811
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Mon Sep 19 22:17:26 2022 +0300

    domd, domu: add pci=pcie_bus_perf parameter for kernels

    Option "pci=pcie_bus_perf" somehow fixes issues with NVME SSDs, when
    writing/reading SSD resulted in corrupted data. This workaround was
    discovered by Renesas.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 19e32fc9cce0697c7ca6721ee81106efa289fbba
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Mon Sep 19 22:12:05 2022 +0300

    domd, domu: install e2fsprogs onto rootfs

    To facilitate testing, install e2fsprogs package, which includes
    mkfs.ext[234] tools. This will ease up testing of PCIe NVME drives.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 65ece1da2643fed8e58549223722761467106666
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Wed Aug 31 23:32:52 2022 +0300

    README.md: Update readme for release 0.8.4

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 1cd8783ff1be560ca850b540e5740b92bc309366
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Wed Aug 31 23:32:05 2022 +0300

    domd, domu: linux: advance linux kernel revision

    Move SRCREV forward, to ensure that latest networking fixes are
    present in the next release.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit b90abf6658666b736882cb209e8413bf38361b1b
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Wed Aug 31 23:31:31 2022 +0300

    dom0: kernel: advance kernel version to 5.10.41/5.1.7.rc3

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit ca2e97f42956abab132bec24f12f7a7e73e8e6b0
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Mon Jul 25 22:35:47 2022 +0300

    README: Update README.md for the 0.8.3 release

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit e12f1f6dd8c6c104cbe363ae178abf447e54a436
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Mon Jul 25 22:32:34 2022 +0300

    domd, domu: update linux kernel to 5.1.7.rc2 version

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 3f8ec4ce39852a7d358c45e8ffdca3cdcdcd7acc
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Mon Jun 27 15:46:07 2022 +0300

    domd, domu: install iperf3 package

    It is used to measure networking performance.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 5d2d790ba1702f37bd177504b378a7a1a19b9d6d
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Mon Jun 27 15:37:40 2022 +0300

    optee: disable hardware features, provide mutex delay

    Disable HW features as they are not fully compatible with the
    virtualization, provide CFG_RCAR_MUTEX_DELAY value, as it is
    expected by new Renesas OP-TEE release.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit d43335f174124203bba6855ae2e4be70e65ce499
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Mon Jun 27 15:37:10 2022 +0300

    domd, domu: linux: compile dmatest as module

    This is expected by latest Renesas release.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 8cdecfc2a66a99d1aefe7ed91d462572ce722429
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri Jun 24 19:49:30 2022 +0300

    domx: optee: remove already applied patch

    Patch 0002-mk-gcc.mk-return-back-CXX-sm-definition.patch was applied
    to upstream, so we need to remove it from this layer.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 9414c187931a9e12fce992d529a6ca71ad3287d1
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri Jun 24 18:37:50 2022 +0300

    README.md: Update readme for 0.8.2 release

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit f42c51d1d36228d358be089a24d6ccd76ffa4eae
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri Jun 24 18:37:23 2022 +0300

    xen: advance xen revision to spider-0.8.2 branch

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit d412a3a54d6d6117193486b17fbdcae21df49b3e
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri Jun 24 19:17:54 2022 +0300

    dom0: force boot network interface for DomD

    As there are all TSN0, TSN1, TSN2 interface possibly working,
    force DomD to use tsn0 for boot.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 70ce8c40df7512406fe282575489794d6b75bd77
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri Jun 24 18:34:04 2022 +0300

    domd: compile Marvell 10G PHY driver into kernel

    Prior to this patch, Marvell 10G PHY driver was built as a module,
    which is not good, as we need this module to properly configure
    network interfaces during NFS boot.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 7cd172f975b9cb1beb7bf90d5a2cfc09d1433e94
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu Jun 23 22:25:02 2022 +0300

    dom0: domu: pass-through tsn1 to DomU

    Change domu configuration file, so we will pass tsn1 interface
    instead of purely virtual vmq0.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 6ec61fe14d4685743498da2f052df0ab1e35e805
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu Jun 23 22:23:08 2022 +0300

    domd: describe tsn1 network interface

    Tell systemd-networkd about tsn1 interface, so it will bring it up.
    This is all what we need from systemd, as this interface then will
    be passed to DomU.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 41df0ca7dfce405db26a955f73b9b996e10291bc
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu Jun 23 22:19:46 2022 +0300

    domu: add tsn1.network file

    We are going to pass tsn1 to DomU. Make sure that DomU knows what
    to do with the new network device.

    In this case we will enable DHCP client for it.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit b12e62239364b70fe03e6b9f285e0021c12f8416
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu Jun 23 22:17:14 2022 +0300

    domd, domu: linux: update revision

    More linux kernel revision forward. Now rswitch back and front will
    support TSN pass-through.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit f8d6180cdce95f6d983e15fd7e6e6367d42547a1
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri May 27 21:30:00 2022 +0300

    yaml: update poky revision to latest version

    Update to Poky to latest revision of Dunfell branch

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 29e070363020aec3522dcc22bd19e5dae7426b72
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri May 27 21:08:45 2022 +0300

    README.md: Update U-Boot environment for PCIe init

    Add another i2c command. This one switches to GPIO422 high output
    mode, which is required to enable certain PCIe devices.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 57c78088b99f45e2dc151f2f6acb2da7c66d0a0e
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri May 27 15:26:02 2022 +0300

    README: fix mistake in u-boot environment

    Add missed `run` command for `bootcmd_tftp`.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 74c73ef5591167e5f1bea4da24866bfa1036f111
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri May 27 15:23:10 2022 +0300

    dom0: change default vPCI device for domu

    To ease up SR-IOV testing, change default PCI SBFD to function 2,
    which is first Samsung's SSD SR-IOV virtual function.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit a557168914169c597f9e0aa810c903c1c79240a3
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri May 27 15:21:59 2022 +0300

    domd: u-boot: upgrade to version rcar-5.1.1-rc2

    This patch should be removed when Renesas updated their base
    layer.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 77c03a5cb0ade037b234d767578f64e866e694e5
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Fri May 27 13:37:27 2022 +0300

    domu: linux: upgrade to 5.1.6-rc3 kernel

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 763d4bc582788f6cc77629a684babb21b536bcaf
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu May 26 22:50:25 2022 +0300

    domd: linux: upgrade to 5.1.6-rc3 kernel

    Change kernel branch and rebase all local patches to the new kernel
    version.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 5d43d581b7cf8fb9310a1a7d4f0638c77dbb1531
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu May 26 21:42:44 2022 +0300

    README: update readme for 0.8.1 release

    Update versions, describe how to configure SR-IOV feature on
    NVME SSD. Samsung PM1733 SSD used as example device.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 486d5d94ade8d61c5a2b03810f20c08520b75ff4
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu May 26 21:40:10 2022 +0300

    domd: xen.dts: add one more IP-MMU context for PCIe0

    Add context for BFD 1:0:2 and onward to facilitate SR-IOV virtual
    functions of Samsung PM1733 SSD.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 6820197ca88cecfc0969f96dbef2ab0b483f1937
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu May 26 21:39:17 2022 +0300

    domd: linux: emulate reading from ECAM under Xen

    If kernel is running as Xen guest, hypervisor will emulate ECAM
    spaces for us. Thus, we need to replace child_ops with ops
    that access ECAM address space.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 09e5822146dcb15b3eee283234ffe90e1bc7d2be
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu May 26 21:37:44 2022 +0300

    domd: linux: enable GPIO_SYSFS module

    This module is mostly needed to configure PCIe-related pins.
    Also the same can be achieved with u-boot i2c commands (see README).

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 3c354d6e4c5183f9d6f78d6955a24cb4985ac371
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu May 26 21:36:22 2022 +0300

    domd: install nvme-related modules and tools

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit ad379a111b646244bc8588dae4d89c32dcffc722
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu May 26 21:36:02 2022 +0300

    dom0: linux: update kernel to 5.1.6.rc3 version

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

commit 65f03498210a547b3b21cbfea504572d377ff0cd
Author: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date:   Thu May 26 21:34:40 2022 +0300

    xen: switch xen to spider-0.8.1 branch

    Update to new xen branch, as part of releasing activities.

    Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

Eabramov84 commented on aos-rcar-gen4.yaml:
```diff
@@ -44,10 +44,10 @@ common_data:
     - type: git
       url: "https://github.com/xen-troops/meta-xt-common.git"
-      rev: master
+      rev: e27c46b
```
add " " to revisions now moulun will not accept

al1img committed:
[yaml] Add quotes to all rev fields

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[domd] Add rule to forward traffic between xenbr0 and eth0

New meta-aos disables all forward rules by default for security reason. We have
to enable explicit forward rules.

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[yaml] Add AOS_VIS_DATA_PROVIDER to configure VIS data provider

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[yaml] Remove setting AOS_RUNNER variable

No need to set AOS_RUNNER in yaml it is set to crun by default in meta-aos.

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img committed:
[README.md] Change code block `sh` to `bash` as `sh` is not supported

Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>
